src/lib/number.jl

```diff
@@ -28,6 +28,7 @@ for (M, f, arity) in DiffRules.diffrules()
   end
 end
 
+@adjoint Base.literal_pow(::typeof(^), x::Float16, ::Val{p}) where {p} = x^p, Δ -> (nothing, p * x ^ (p - 1), nothing)
```

Comment:
If you're going to add an explicit adjoint for `literal_pow`, which wasn't there before, add it for all `Real`.

(I was imagining that we'd just add the forward definition `literal_pow(::Float16) = x*x` and let Zygote differentiate it, adding the adjoint works well too).